dccrg.hpp

```diff
@@ -3013,19 +3013,19 @@ template <
 			return ret_val;
 		}
 
-		const std::vector<uint64_t>& neighbors_ref
+		const std::vector<std::pair<uint64_t,std::array<int, 4>>>& neighbors_ref
```

Comment:
So I don't get how what landed as a `vector<uint64_t>` now ends up being this vector of pairs with exactly the same rhs.

Reply:
Quite baffling indeed. I can only surmise there was some sort of implicit conversion going on?

Reply:
Afaik C++ doesn't do implicit conversion of reference variables so old version shouldn't compile, seems like it just went unnoticed until now since mainline version doesn't use this function anywhere. I would change the type to auto& though, compiler knows anyway what types are involved...

Reply:
Well the old version was being compiled and used by us with compilation using e.g. g++. But good to know this isn't an issue in mainline DCCRG. Perhaps we can converge again at some point in the (near?) future.

Reply:
Old version of this function? Since templates are involved unused broken code might not trigger compiler errors. Strange if this function did compile previously. Which version of g++ worked?

Reply:
> converge again

If I recall correctly, array<int,4> version stores neighbor info as offsets in sizes of cell itself + that size in grid indices but turned out that this way leaves ambiguities and doesn't provide enough information for some solver(s). Hence mainline now stores neighbor offsets in grid indices and so uses array<int,3> everywhere (internally?). Public interface now uses named variables x,y,z as e.g. shown in advection solver that supports AMR: https://github.com/fmihpc/dccrg/blob/master/tests/advection/solve.hpp#L76